packages/node/src/integrations/tracing/nest.ts

```diff
@@ -44,7 +54,140 @@ interface MinimalNestJsApp {
+ */
+export class SentryNestInstrumentation extends InstrumentationBase {
+  public static readonly COMPONENT = '@nestjs/common';
+  public static readonly COMMON_ATTRIBUTES = {
+    component: SentryNestInstrumentation.COMPONENT,
+  };
+
+  public constructor(config: InstrumentationConfig = {}) {
+    super('sentry-nestjs', '1.0.0', config); // TODO: instrumentationVersion
+  }
+
+  /**
+   * Initializes the instrumentation by defining the modules to be patched.
+   */
+  public init(): InstrumentationNodeModuleDefinition {
+    const moduleDef = new InstrumentationNodeModuleDefinition(SentryNestInstrumentation.COMPONENT, supportedVersions);
+
+    moduleDef.files.push(this._getInjectableFileInstrumentation(supportedVersions));
+    return moduleDef;
+  }
+
+  /**
+   * Wraps the @Injectable decorator.
+   */
+  private _getInjectableFileInstrumentation(versions: string[]): InstrumentationNodeModuleFile {
+    return new InstrumentationNodeModuleFile(
+      '@nestjs/common/decorators/core/injectable.decorator.js',
+      versions,
+      (moduleExports: { Injectable: InjectableTarget }) => {
+        if (isWrapped(moduleExports.Injectable)) {
+          this._unwrap(moduleExports, 'Injectable');
+        }
+        this._wrap(moduleExports, 'Injectable', this._createWrapInjectable());
+        return moduleExports;
+      },
+      (moduleExports: { Injectable: InjectableTarget }) => {
+        this._unwrap(moduleExports, 'Injectable');
+      },
+    );
+  }
+
+  /**
+   * Creates a wrapper function for the @Injectable decorator.
+   *
+   * Wraps the use method to instrument nest class middleware.
+   */
+  private _createWrapInjectable() {
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    return function wrapInjectable(original: any) {
+      return function wrappedInjectable(options?: unknown) {
+        return function (target: InjectableTarget) {
+          // ensure class has not been patched before
+          if (target.sentryPatched) {
+            return original(options)(target);
+          } else {
+            Object.defineProperty(target, sentryPatched, { value: true });
+          }
+
+          // patch middleware
+          if (typeof target.prototype.use === 'function') {
+            const originalUse = target.prototype.use;
+
+            target.prototype.use = function (req: unknown, res: unknown, next: (error?: unknown) => void): void {
+              const prevSpan = getActiveSpan();
+
+              startSpanManual(
+                {
+                  name: target.name,
+                  attributes: {
+                    [SEMANTIC_ATTRIBUTE_SENTRY_OP]: 'middleware.nestjs',
+                    [SEMANTIC_ATTRIBUTE_SENTRY_ORIGIN]: 'auto.middleware.nestjs',
+                  },
+                },
+                (span: Span) => {
+                  // patch  next to end span before next middleware is being called
+                  const wrappedNext = (error?: Error | unknown): void => {
```

Comment:
h: We should forward all arguments not just the first one. Ideally we even use a proxy like for example here: https://github.com/getsentry/sentry-javascript/blob/03257e09cf959a79ecc0a142e2946ebec53e760e/packages/nextjs/src/common/wrapServerComponentWithSentry.ts#L37

Reply:
using a proxy now